.github/workflows/test.yml

```diff
@@ -1,28 +1,49 @@
 name: Tests
 on: [push, pull_request]
 env:
-  NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
-  NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+  CARGO_TERM_COLOR: always
 jobs:
-  check-sha256-example:
+  test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - run: cargo test
-      - run: cargo run --release --example sha256
-  check-sha256-example-wasm:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v3
-    - name: Install Rust
-      uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-        override: true
-        target: wasm32-unknown-unknown
-    - name: Check Build
-      uses: actions-rs/cargo@v1
-      with:
-        command: check
-        args: --target wasm32-unknown-unknown
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Cache cargo build
+        uses: Swatinem/rust-cache@v1
+
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          profile: minimal
+          components: rustfmt, clippy
+          target: wasm32-unknown-unknown
+
+      - name: Run cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+      - name: Run cargo clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --all-targets --all-features -- -D warnings
```

Comment:
Hmm actually, I think we have to avoid `all-features` here, since `llvm` will enforce that a suitable version of llvm is downloaded
```
➜  circom-scotia git:(main) ✗ cargo clippy --all-targets --all-features -- -D warnings
    Checking llvm-sys v140.1.2
error: No suitable version of LLVM was found system-wide or pointed
              to by LLVM_SYS_140_PREFIX.
       
              Consider using `llvmenv` to compile an appropriate copy of LLVM, and
              refer to the llvm-sys documentation for more information.
       
              llvm-sys: https://crates.io/crates/llvm-sys
              llvmenv: https://crates.io/crates/llvmenv
```